### ISSUE TITLE
Add rule to detect non-public property access via element access

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#375]
```ts
class Foo {
    private bar = 1;
}
let foo = new Foo();

foo.bar; // type error
foo['bar']; // can access private property
```

Although `prefer-dot-notation` disallows element access where possible, but there are still property names like `'foo-bar'` that need to use element access. TypeScript intentionally doesn't check accessibility on element access for reasons ... (see https://github.com/Microsoft/TypeScript/issues/26328)

This is likely an error and should be detected by a lint rule

**Comment by ajafff** [comment]
Depending on how https://github.com/Microsoft/TypeScript/issues/26356 and https://github.com/Microsoft/TypeScript/issues/26355 are resolved, this rule should detect these cases, too.

**Comment by ajafff** [comment]
It turns out there are a lot more checks that are simply ignored for element access:
* `super['abstractMethod']`
* `super['prop']`
*  `this['abstractProp']` in constructor
* `a['protectedProp']`: accessing protected property declared in base class on an instance of a base class in a static method of a derived class (or a function with a `this` parameter of a derived class)
